src/build.d

```diff
@@ -1548,6 +1554,7 @@ class BuildRule
     **/
     bool run(bool depUpdated = false)
     {
+        assert(!name.empty, "Build " ~ msg ~ " has no name");
```

Comment:
I didn't want to add it as an (expensive) invariant as we only have a handful of static rules.